src/Interro.java

```diff
@@ -155,32 +148,218 @@ public void paintComponent(Graphics g){
+                        }
+                    }
+                    //tree leaves
+                    for(int i = 0; i < trees.length; i++) {
+                        for (int j = 0; j < 2; j++) {
+                            g2.setColor(new Color(22 / 255f, 128 / 255f, 48 / 255f, 0.5f));
+                            g2.fillOval((trees[i][0] - 3 - pos[0]) * scale / scope + 500 - scale / scope / 2, (trees[i][1] - 3 - pos[1]) * scale / scope + 500 - scale / scope / 2, 12 * scale / scope, 12 * scale / scope);
+                        }
+                    }
+                    //reference lines
+                    g2.setColor(Color.BLACK);
+                    g2.drawLine(500, 0, 500, 1000);
+                    g2.drawLine(0, 500, 1000, 500);
+                }
+                if(f3){
+                    g2.setColor(Color.BLACK);
+                    g2.setFont(new Font("Serif", Font.PLAIN, 30));
+                    g2.drawString("x: "+pos[0]+", y: "+pos[1], 10, 40);
+                }
+            }
+
+            class PlayerMover implements ActionListener{
+                public void actionPerformed(ActionEvent e){
+                    if(move == 2){
+                        move = 0;
+                        if (up && pos[1] > 0) pos[1] -= speed;
+                        if (down && pos[1] < 1000) pos[1] += speed;
+                        if (left && pos[0] > 0) pos[0] -= speed;
+                        if (right && pos[0] < 1000) pos[0] += speed;
+                    }
+                    else move++;
+                }
+            }
+
+            class Render implements ActionListener{
+                public void actionPerformed(ActionEvent e){
+                    grabFocus();
+                    repaint();
+                }
+            }
+
+            public void keyTyped(KeyEvent e){
+                char key = e.getKeyChar();
+                switch (key){
+                    case 'm': {
+                        map = !map;
+                        break;
+                    }
+                    case '/': {
+                        f3 = !f3;
+                        break;
+                    }
+                }
+            }
+
+            public void keyPressed(KeyEvent e){
+                char key = e.getKeyChar();
+                switch(key){
+                    case 'w': {
+                        up = true;
+                        break;
+                    }
+                    case 's': {
+                        down = true;
+                        break;
+                    }
+                    case 'a': {
+                        left = true;
+                        break;
+                    }
+                    case 'd': {
+                        right = true;
+                        break;
+                    }
```

Comment:
For each of these `case`s, should the fields not being set to `true` be set to `false`? Doing so might be important if this class is "stateful" (ie it remembers the states of its fields rather than objects being created from scratch each time). I'm guessing it's stateful since that's the default way Java programming is learned. The alternative is "stateless" which means each objects' state (ie its fields) don't change once it's initialized. This makes concurrent programming (eg with threads) easier since locks aren't needed to use the object.